src/package.cc

```diff
@@ -62,4 +62,79 @@ Package::Package(std::string name, double fill_min, double fill_max,
   }
 }
 
+TransportUnit::Ptr TransportUnit::unrestricted_ = NULL;
+
+TransportUnit::Ptr TransportUnit::Create(std::string name, int fill_min, int fill_max, std::string strategy) {
+  if (fill_min < 0 || fill_max < 0) {
+    throw ValueError("fill_min and fill_max must be non-negative");
+  }
+  else if (fill_min > fill_max) {
+    throw ValueError("fill_min must be less than or equal to fill_max");
+  }
+  Ptr t(new TransportUnit(name, fill_min, fill_max, strategy));
+  return t;
+}
+
+// singleton pattern: 
+// if the static member is not yet set, create a new object
+// otherwise return the object that already exists
+TransportUnit::Ptr& TransportUnit::unrestricted() {
+  if (!unrestricted_) {
+    unrestricted_ = Ptr(new TransportUnit(unrestricted_name_));
+  }
+  return unrestricted_;
+}
+
+int TransportUnit::GetTransportUnitFill(int qty) {
+  if (qty < fill_min_) {
+    // less than one pkg of material available
+    return 0;
+  }
+
+  if (strategy_ == "first") {
+    return fill_max_;
+  } else if (strategy_ == "equal") {
+    // int division automatically rounds down. don't need floor in min, and 
+    // get ceil by hand instead 
+    int num_at_min_fill = qty / fill_min_;
+    int num_at_max_fill = (qty + fill_max_ - 1) / fill_max_;
+
+    if (num_at_min_fill >= num_at_max_fill) {
+      // all material *might* fit transport units. However, this is more
+      // challenging than packages because transport units are discrete. 
+      double dbl_fill_mass = (double)qty / (double)num_at_max_fill;
+      return std::floor(dbl_fill_mass);
+    }
+    // some material will remain unrestricted, fill up as many transport
+    // units as possible. Or, perfect fill is possible but not with integer
+    // fill (see above). 
+  }
+  return fill_max_;
+}
+
+int TransportUnit::MaxShippablePackages(int pkgs) {
+  int TU_fill;
+  int shippable = 0;
+
+  if (pkgs == 0 && pkgs < fill_min_) {
+    return 0;
+  }
+
+  while ((pkgs > 0) && (pkgs >= fill_min_)) {
```

Comment:
Which strategy is this?

Reply:
Now that I think about it, this is basically a new iterative strategy that's more efficient than `equal` or `first` specifically for `TransportUnit`s because they're restricted to integers. Maybe I'll call it `hybrid`, and it should probably be the default? Does come with more work, because it iterates over each transport-unit-size collection of packages, but also is able to ship more packages that the other strategies. It's not your `balanced` proposal above, but it's still a more flexible option

Example:
- transport unit with minimum 3 and maximum 4
- 11 packages
- Equal or first are gonna give you a fill of 4, which would only allow 8 packages to be sent. However hybrid will fill 2 transport units, see that there are 3 remaining,and fill a final transport unit with 3 allowing all 11 packages to be shipped